### DISCUSSION TITLE
How to filter and display data from the JSON array

### DISCUSSION THREAD

**Discussion by oleksandr-zelenchuk** [grafana/grafana-infinity-datasource#813] [Q&A]
Hello,
Could you please help me to filter the data and display it in the dashboard?
For example, I receive a JSON array of HTTP statuses from Cloudflare using GraphQL, from I need to display only a few of them in the dashboard. 
**Here is an array:** 
[{"edgeResponseStatus":200,"requests":565},{"edgeResponseStatus":204,"requests":7},{"edgeResponseStatus":301,"requests":45},{"edgeResponseStatus":302,"requests":66},{"edgeResponseStatus":303,"requests":4},{"edgeResponseStatus":304,"requests":34},{"edgeResponseStatus":400,"requests":11},{"edgeResponseStatus":401,"requests":2},{"edgeResponseStatus":403,"requests":27},{"edgeResponseStatus":404,"requests":7},{"edgeResponseStatus":405,"requests":11},{"edgeResponseStatus":409,"requests":1},{"edgeResponseStatus":422,"requests":1},{"edgeResponseStatus":429,"requests":7},{"edgeResponseStatus":499,"requests":9},{"edgeResponseStatus":500,"requests":1}]

As I checked, it cannot be done with Rows and Selector, because in this case the data is not displayed at all. Computed columns are also not an option.
![2024-04-06_14-06](https://github.com/grafana/grafana-infinity-datasource/assets/63159500/f8a44c44-2ffd-4c4b-9dfd-f447fbbdca00)

**The query is the following:**

> query {
  viewer {
    zones(filter: { zoneTag: "$zoneTag" }) {
      httpRequests1mGroups(limit: 20, filter: {datetime_geq: "${__from:date}", datetime_lt: "${__to:date}"}, orderBy: [datetime_ASC]) {
        dimensions {
          datetime
        }
        sum {
          responseStatusMap {
            requests
            edgeResponseStatus
          }  
        }
      }
    }
  }
}

Does Infinity have a plugin, some internal tool for this filtering?

Thank you in advance.

**Comment by yesoreyeram** [comment]
See if https://grafana.com/docs/plugins/yesoreyeram-infinity-datasource/latest/query/filters/ helps

**Reply by oleksandr-zelenchuk** [reply]
Unfortunately, these options are not valid for the current case since the array cannot be parsed by the key value. Because it is the same for every pair ("edgeResponseStatus").

For example, this is how it looks for me:
![2024-04-23_21-13](https://github.com/grafana/grafana-infinity-datasource/assets/63159500/358d7b84-9dd4-4d02-aee4-f7a0ddb2cb51)


Is there any way to do it using the Infinity plugin?
